provisioning/service/src/Config/EnrollmentGroup.cs

```diff
@@ -0,0 +1,313 @@
+        /// <param name="provisioningStatus">the <see cref="ProvisioningStatus"/> that determine the initial status of the device. This is optional and can be <code>null</code>.</param>
+        /// <param name="createdDateTimeUtc">the <code>DateTime</code> with the date and time that the enrollment was created. This is optional and can be <code>null</code>.</param>
+        /// <param name="lastUpdatedDateTimeUtc">the <code>DateTime</code> with the date and time that the enrollment was updated. This is optional and can be <code>null</code>.</param>
+        /// <param name="eTag">the <code>string</code> with the eTag that identify the correct instance of the enrollment in the service. It cannot be <code>null</code> or empty.</param>
+        /// <exception cref="ProvisioningServiceClientException">if the received JSON is invalid.</exception>
+        [JsonConstructor]
+        public EnrollmentGroup(
+            string enrollmentGroupId,
+            AttestationMechanism attestation,
+            string iotHubHostName,
+            TwinState initialTwinState,
+            ProvisioningStatus provisioningStatus,
+            DateTime createdDateTimeUtc,
+            DateTime lastUpdatedDateTimeUtc,
+            string eTag)
+        {
+            /* SRS_ENROLLMENT_GROUP_21_003: [The constructor shall throws ProvisioningServiceClientException if one of the 
+                                                    provided parameters in JSON is not valid.] */
+            if (string.IsNullOrWhiteSpace(eTag))
+            {
+                throw new ProvisioningServiceClientException("Service respond an enrollmentGroup without eTag.");
+            }
+            if(attestation == null)
+            {
+                throw new ProvisioningServiceClientException("Service respond an enrollmentGroup without attestation.");
+            }
+            try
+            {
+                /* SRS_ENROLLMENT_GROUP_21_004: [The constructor shall store all parameters in the JSON.] */
+                EnrollmentGroupId = enrollmentGroupId;
+                Attestation = attestation.GetAttestation();
+                IotHubHostName = iotHubHostName;
+                InitialTwinState = initialTwinState;
+                ProvisioningStatus = provisioningStatus;
+                CreatedDateTimeUtc = createdDateTimeUtc;
+                LastUpdatedDateTimeUtc = lastUpdatedDateTimeUtc;
+                ETag = eTag;
+            }
+            catch (ArgumentException e)
+            {
+                throw new ProvisioningServiceClientException(e);
+            }
+        }
+
+        /// <summary>
+        /// Convert this object in a pretty print format.
+        /// </summary>
+        /// <returns>The <code>string</code> with the content of this class in a pretty print format.</returns>
+        public override string ToString()
+        {
+            string jsonPrettyPrint = Newtonsoft.Json.JsonConvert.SerializeObject(this, Formatting.Indented);
```

Comment:
Please avoid fully qualified object names. (See the coding guidelines example `using` statements.)

Reply:
Sure, VS put it... removed.